### PR TITLE
Use ternary operator in a series of if statements to simplify

### PR DESCRIPTION
**Pull request by DanVanAtta**

### CODE REVIEW

File: src/main/java/games/strategy/triplea/delegate/BattleCalculator.java [hunk 1]
```diff
@@ -1339,38 +1339,27 @@ public static int getUnitPowerForSorting(final Unit current, final boolean defen
...
+        final int tempStrength = (defending) ? ua.getDefense(current.getOwner()) :
+            ua.getAttack(current.getOwner());
         strengthWithoutSupport +=
             TerritoryEffectHelper.getTerritoryCombatBonus(current.getType(), territoryEffects, defending);
         strengthWithoutSupport += Math.min(Math.max(tempStrength, 0), data.getDiceSides());
       }
     }
     return strengthWithoutSupport;
-  }
+}
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/BattleCalculator.java]: Formatting

File: src/main/java/games/strategy/triplea/delegate/BattleCalculator.java [hunk 2]
```diff
@@ -1339,38 +1339,27 @@ public static int getUnitPowerForSorting(final Unit current, final boolean defen
...
       strengthWithoutSupport = Math.min(Math.max(strengthWithoutSupport + 1, 0), data.getDiceSides());
     } else {
       for (int i = 0; i < rolls; i++) {
-        final int tempStrength;
-        if (defending) {
-          tempStrength = ua.getDefense(current.getOwner());
-        } else {
-          tempStrength = ua.getAttack(current.getOwner());
-        }
+        final int tempStrength = (defending) ? ua.getDefense(current.getOwner()) :
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/BattleCalculator.java]: Parentheses are not needed around `defending`

File: src/main/java/games/strategy/triplea/delegate/BattleCalculator.java [hunk 3]
```diff
@@ -1339,38 +1339,27 @@ public static int getUnitPowerForSorting(final Unit current, final boolean defen
...
     int strengthWithoutSupport = 0;
     // Find the strength the unit has without support
     // lhtr heavy bombers take best of n dice for both attack and defense
     if (rolls > 1 && (lhtrBombers || ua.getChooseBestRoll())) {
-      if (defending) {
-        strengthWithoutSupport = ua.getDefense(current.getOwner());
-      } else {
-        strengthWithoutSupport = ua.getAttack(current.getOwner());
-      }
+      strengthWithoutSupport = (defending) ? ua.getDefense(current.getOwner()) :
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/BattleCalculator.java]: Parentheses are not needed around `defending`

File: src/main/java/games/strategy/triplea/delegate/BattleCalculator.java [hunk 4]
```diff
@@ -1339,38 +1339,27 @@ public static int getUnitPowerForSorting(final Unit current, final boolean defen
       final Collection<TerritoryEffect> territoryEffects) {
     final boolean lhtrBombers = games.strategy.triplea.Properties.getLHTR_Heavy_Bombers(data);
     final UnitAttachment ua = UnitAttachment.get(current.getType());
-    final int rolls;
-    if (defending) {
-      rolls = ua.getDefenseRolls(current.getOwner());
-    } else {
-      rolls = ua.getAttackRolls(current.getOwner());
-    }
+    final int rolls = (defending) ?  ua.getDefenseRolls(current.getOwner()) :
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/BattleCalculator.java]: Parentheses are not needed around `defending`

Comment (DanVanAtta) [inline src/main/java/games/strategy/triplea/delegate/BattleCalculator.java]: Indeed. Adding them is to  make the ternary operator more explicit and is a bit of a stylistic thing. The non-paren case I found less readable.  
Ternary operators can be nice, the optional parens I believe are often recommended to keep the conditional part of the syntax a bit more explicit.

**Comment by DanVanAtta** [comment]
@RoiEXLab  the optional parens are intentional, they improve readability here IMO

**Pull request merged**